grepos/app/src/main/java/net/boondockradio/grepos/adapter/RepositoryAdpter.java

```diff
@@ -3,47 +3,121 @@
 import net.boondockradio.grepos.R;
 import net.boondockradio.grepos.dto.Repository;
 
+import android.support.v7.widget.LinearLayoutManager;
 import android.support.v7.widget.RecyclerView;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
+import android.widget.ProgressBar;
 import android.widget.TextView;
 
 import java.util.List;
 
-public class RepositoryAdpter extends RecyclerView.Adapter<RepositoryAdpter.ItemViewHolder> {
+public class RepositoryAdpter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
+    private final int VIEW_ITEM = 1;
+    private final int VIEW_PROGRESS = 0;
+
+    private final int LOAD_MORE_THRESHOLD = 2;
 
     private List<Repository> mRepositoryItems;
+    private RecyclerView mRecyclerView;
+
+    private OnLoadMoreListener onLoadMoreListener;
+    private int total = 0;
+    private int last = 0;
+    private boolean isLoading = false;
 
-    public RepositoryAdpter(List<Repository> repositoryItems) {
+    public RepositoryAdpter(List<Repository> repositoryItems, RecyclerView recyclerView) {
         mRepositoryItems = repositoryItems;
+        mRecyclerView = recyclerView;
+
+        final LinearLayoutManager layoutManager = (LinearLayoutManager) mRecyclerView.getLayoutManager();
```

Comment:
We should use the constructor to `initialize` an object.
It shouldnt do any `work`